### PR TITLE
Added `TxPointer` to `Mint` transaction

### PR DESCRIPTION
**Pull request by xgreenx**
[Preview](https://github.com/FuelLabs/fuel-specs/blob/feature/mint-tx-pointer/specs/protocol/tx_format.md#transactionmint)

During the implementation of https://github.com/FuelLabs/fuel-tx/pull/202 we found that we need a way to make `Mint` transactions unique. After discussion on the issue and PR, we decided to add the `TxPointer` field.

Close https://github.com/FuelLabs/fuel-specs/issues/421

### CODE REVIEW

File: specs/protocol/tx_format.md
Comment (Voxelot) [inline specs/protocol/tx_format.md:174]:
```suggestion
It's not usable outside of block creation or execution, so all fields must be fully set upon creation without any zeroing.
```

**Comment by Voxelot** [comment]
lgtm, waiting for @adlerjohn approval

File: specs/protocol/tx_format.md
Comment (adlerjohn) [inline specs/protocol/tx_format.md:417]:
```suggestion
The location of the transaction in the block. It can be used by UTXOs as a reference to the transaction or by the transaction itself to make it unique.
```

File: specs/protocol/tx_format.md
Comment (adlerjohn) [inline specs/protocol/tx_format.md:174]:
```suggestion
The transaction is created by the block producer and is not signed. Since it is not usable outside of block creation or execution, all fields must be fully set upon creation without any zeroing.
```

File: specs/protocol/tx_format.md
Comment (adlerjohn) [inline specs/protocol/tx_format.md:171]: Woops. This change should make the ToC above proper.

```suggestion
### TransactionMint
```

File: specs/protocol/tx_format.md
Comment (adlerjohn) [inline specs/protocol/tx_format.md:178]: Will require re-formatting the table I think.

```suggestion
| `txPointer`     | [TXPointer](#txpointer) | The location of the `Mint` transaction in the block. |
```

**Review by adlerjohn — COMMENTED** [review]
formatting nits

**Review by Voxelot — APPROVED** [review]
approving since John's requested changes were pushed

**Review by adlerjohn — APPROVED** [review]
:shipit:

**Pull request merged**
